Add Test containers for test cases

rtclauss opened the pull request:
Note that some tests may fail. If you want to skip the tests add `-DskipTests=true` as part of your build.

Also, I've added a startup bean in 7b54dcf943e17670897c78a6197143c3712bd649 which creates the necessary indexes in CouchDB to improve performance when there is a lot of content in the CouchDB.

Commit:
Update dependencies to latest

Commit:
Create CouchDB index at startup to improve speed

Commit:
Update classes to use correct JNosql classes

Commit:
Add test cases

Commit:
Add test case documentation

jwalcorn reviewed (commented):
In general, I like it - thanks for adding these tests, and automatically making that index.  I do see some random line of output in the readme where it shows how to invoke stuff (line 66), that should be cleaned up.  Also not sure why the mention of the option to run CouchDB in a VM in the cloud got removed from the readme; if I use the Azure portal to request a CouchDB deployment, it is indeed a VM in the cloud.  Lastly, unless I'm misunderstanding stuff, I don't see it actually checking if CouchDB is available in the CouchDBReadinessTest (it seems to be the same as the NoCouchDBReadinessTest)

rtclauss commented:
I added additional clarification to the README.md.

The difference between `CouchDBReadinessTest` and `NoCouchDBReadinessTest` is that the first test scenario starts up CouchDB testcontainer, then tests the health check endpoint to verify that the health check has passed while the second scenario *does not*  start CouchDB and verifies that the health check fails.

I've updated the Javadocs in the most recent commit.

Commit:
update javadoc for testcases

Commit:
Update readme with more details on building the container

Commit:
Clean up extra wingdings

jwalcorn reviewed (approved):
Looks good now.  Thanks!
